x-pack/plugins/security/server/config.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { schema, Type, TypeOf } from '@kbn/config-schema';
+import { PluginInitializerContext } from '../../../../src/core/server';
+
+export type ConfigType = ReturnType<typeof createConfig$> extends Observable<infer P>
+  ? P
+  : ReturnType<typeof createConfig$>;
+
+const providerOptionsSchema = (providerType: string, optionsSchema: Type<any>) =>
+  schema.conditional(
+    schema.siblingRef('providers'),
+    schema.arrayOf(schema.string(), {
+      validate: providers => (!providers.includes(providerType) ? 'error' : undefined),
+    }),
+    optionsSchema,
+    schema.never()
+  );
+
+export const ConfigSchema = schema.object(
+  {
+    cookieName: schema.string({ defaultValue: 'sid' }),
+    encryptionKey: schema.conditional(
+      schema.contextRef('dist'),
+      true,
+      schema.maybe(schema.string({ minLength: 32 })),
+      schema.string({ minLength: 32, defaultValue: 'a'.repeat(32) })
+    ),
+    sessionTimeout: schema.oneOf([schema.number(), schema.literal(null)], { defaultValue: null }),
+    secureCookies: schema.boolean({ defaultValue: false }),
+    public: schema.object({
+      protocol: schema.maybe(schema.oneOf([schema.literal('http'), schema.literal('https')])),
+      hostname: schema.maybe(schema.string({ hostname: true })),
+      port: schema.maybe(schema.number({ min: 0, max: 65535 })),
+    }),
+    // This property is deprecated, but we include it here since new platform doesn't support config deprecation
+    // transformations yet (e.g. `rename`), so we handle it manually for now.
+    authProviders: schema.maybe(schema.arrayOf(schema.string(), { minSize: 1 })),
```

Comment:
__note:__ diff#12: that's the hack, I didn't find a better way to handle this "rename" deprecation use case with the primitives core provides currently. The config deprecations in NP is on immediate Mikhail roadmap, but we're not there yet. Let me know if you have a better idea or just don't to see this hack at all :)